pcsx2/gui/GlobalCommands.cpp

```diff
@@ -150,15 +150,20 @@ namespace Implementations
 			OSDlog( Color_StrongRed, true, "(FrameLimiter) SlowMotion ENABLED." );
 			g_Conf->EmuOptions.GS.FrameLimitEnable = true;
 		}
+
+		GSsetVsync(EmuConfig.GS.GetVsync());
+
 		pauser.AllowResume();
 	}
 
 	void Framelimiter_MasterToggle()
 	{
 		ScopedCoreThreadPause pauser;
 		g_Conf->EmuOptions.GS.FrameLimitEnable = !g_Conf->EmuOptions.GS.FrameLimitEnable;
-		GSsetVsync( g_Conf->EmuOptions.GS.FrameLimitEnable && g_Conf->EmuOptions.GS.VsyncEnable );
 		OSDlog( Color_StrongRed, true, "(FrameLimiter) %s.", g_Conf->EmuOptions.GS.FrameLimitEnable ? "ENABLED" : "DISABLED" );
+
+		GSsetVsync(EmuConfig.GS.GetVsync());
```

Comment:
@mirh I wrote turbo not frame limiter.

Currently when the turbo is enabled, vsync is force disabled. Otherwise the turbo will be limited to refresh rate, which is 60 fps most of the time (so not a turbo). However, on high speed monitor it is doable. So you will be able to run turbo in high quality (i.e. no tearing). I don't know what will happen if the user can't manage the turbo fps target (stuttering maybe if not adaptive vsync).

On variable refresh rate monitor, the vsync parameter must remain off (or always on I don't know) so current behavior is broken.

Then on the framelimiter topic. I think all combinations of framelimiter/vsync are possibles. So it doesn't make sense to interlock them.